src/textual/reactive.py

```diff
@@ -281,36 +328,53 @@ def invoke_watcher(
                         sender=obj, callback=partial(await_watcher, watch_result)
                     )
                 )
-                return False
-            else:
-                return True
 
-        watch_function = getattr(obj, f"watch_{name}", None)
+        # Get the current value.
+        value = self.__get__(obj)
+
+        watch_function = self.watch_function
         if callable(watch_function):
-            invoke_watcher(watch_function, old_value, value)
+            invoke_watcher(partial(watch_function, obj), old_value, value)
 
-        watchers: list[Callable] = getattr(obj, "__watchers", {}).get(name, [])
+        watchers: list[Callable] = getattr(obj, "__watchers", {}).get(self.name, [])
         for watcher in watchers:
             invoke_watcher(watcher, old_value, value)
 
+    ##
+    ## ONCE PER `Reactable` OBJECT INSTANCE
+    ##
+
     @classmethod
-    def _compute(cls, obj: Reactable) -> None:
+    def _initialize_reactable_object(cls, obj: Reactable) -> None:
+        """Set defaults and call any watchers / computes for the first time.
+
+        Args:
+            obj: An object with Reactive descriptors
+        """
+        _rich_traceback_guard = True
+        for reactive in obj._reactives.values():
+            reactive._initialize_reactive(obj)
+
+    @classmethod
+    def _reset_reactable_object(cls, obj: Reactable) -> None:
+        """Reset reactive structures on object (to avoid reference cycles).
+
+        Args:
+            obj: A reactive object.
+        """
+        _rich_traceback_guard = True
+        getattr(obj, "__watchers", {}).clear()
```

Comment:
Notice that the refactoring got rid of `__computes`, which wasn't really used also...